src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -72,6 +74,58 @@ describeEachAppLayout(({ theme, size }) => {
     expect(wrapper.findDrawersTriggers()).toHaveLength(2);
   });
 
+  test('update rendered drawers via runtime config', async () => {
+    awsuiPlugins.appLayout.registerDrawer({ ...drawerDefaults, resizable: true });
+    const { wrapper } = await renderComponent(<AppLayout />);
+    // the 2nd trigger is for tools
+    expect(wrapper.findDrawersTriggers()).toHaveLength(2);
+    const drawerTrigger = wrapper.findDrawerTriggerById(drawerDefaults.id);
+
+    if (theme !== 'classic' && size === 'desktop') {
+      const triggerParent = drawerTrigger?.getElement().parentElement;
+      const triggerParentWrapper = createWrapper(triggerParent!);
+
+      awsuiPlugins.appLayout.updateDrawer({
+        id: drawerDefaults.id,
+        badge: true,
+      });
+      await delay();
+
+      const dotElement = triggerParentWrapper
+        ?.findByClassName((theme === 'refresh' ? triggerStyles : toolbarTriggerStyles).dot)
+        ?.getElement();
+
+      expect(dotElement).toBeInTheDocument();
+
+      awsuiPlugins.appLayout.updateDrawer({
+        id: drawerDefaults.id,
+        badge: false,
+      });
+      await delay();
+
+      expect(dotElement).not.toBeInTheDocument();
```

Comment:
You need to re-query the DOM for a new dot instance. Because even if the previous `dotElement` is no longer there, it could be another one, after re-rendering

Reply:
updated.